### DISCUSSION TITLE
Can a parameter be passed to the DataPortal_Update or [Update] just like during the Fetch?

### DISCUSSION THREAD

**Discussion by drakeforte5** [MarimerLLC/csla#1615] [General]
The fetch allows me to pass parameters.

  [Fetch]
 private void Fetch(string scId, [Inject] DbContextFactory factory)..

but the Update throws an error if I pass a parameter.  I was assuming that if I override the SaveAsync on the BusinessListBase, the userState would be the parameter.

So whenI try to do this:

  [Update]
 private async Task Update(string scId, [Inject] DbContextFactory factory)

I get: Error DataPortal.Update failed (Invalid operation - update not allowed):DataList.DataPortal_Update method call failed

If I remove the scid parameter then the Update works.

Is there a way to pass a parameter on the DP Update?

Thanks.

**Comment by rockfordlhotka** [comment]
There is not a way to pass a parameter to insert/update/delete operations.

What is the scenario where you want to pass a parameter?

**Comment by drakeforte5** [comment]
I have multiple database instances for an app wherein the UI selection from the user drives how I construct my database connection string.  

I was able to get around it by using the Csla.ApplicationContext.Clientcontext.

**Reply by rockfordlhotka** [reply]
Perfect, that is the answer I was going to suggest. Or `LocalContext` or config injected via DI.
